CalorieTracker/CalorieTracker/Controllers/CalorieController.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  CalorieController.swift
+//  CalorieTracker
+//
+//  Created by Sean Acres on 8/23/19.
+//  Copyright © 2019 Sean Acres. All rights reserved.
+//
+
+import Foundation
+import CoreData
+
+class CalorieController {
+    func createCaloriesEntry(amount: String) {
+        guard let amountDouble = Double(amount) else { return }
+        let _ = CaloriesEntry(amount: amountDouble)
+        
+        saveToPersistentStore()
+    }
+    
+    func saveToPersistentStore() {
+        let moc = CoreDataStack.shared.mainContext
+        
+        moc.perform {
```

Comment:
Right now this is the only place from which you're saving data, but I think it might be better to abstract this `moc.perform` method onto the `CoreDataStack` that way you don't have to remember to `perform` or `performAndWait` on every save.

The save method on the `CoreDataStack` could look like this:
```swift
func save(context: NSManagedObjectContext = CoreDataStack.shared.mainContext) throws {
        
        var error: Error?
        
        context.performAndWait {
            do {
                try context.save()
            } catch let saveError {
                error = saveError
            }
        }
        if let error = error { throw error }
    }
```

Then your call to save in this controller wouldn't have to call perform and wait. It would just have to call `CoreDataStack.shared.save()` within a do-try-catch block.

```swift
private func saveToPersistentStore() {
        do {
            try CoreDataStack.shared.save()
            NotificationCenter.default.post(name: .recordsWereUpdated, object: nil)
        } catch {
            NSLog("Error saving moc: \(error)")
        }
    }
```